### ISSUE TITLE
Rethink whether RemotePointEvaluation should live in namespace MPI

### ISSUE THREAD

**Issue by nfehn** [dealii/dealii#13500]
In my opinion, the serial functionality of `RemotePointEvaluation` should be relevant regarding the namespace in which this class lives. I would suggest something like `GridTools` instead of `Utilities::MPI` (`Triangulation` and many other classes also do not live in a namespace `MPI` just because they work in parallel.).

**Comment by drwells** [comment]
I agree - things in `base` are not supposed to depend on things in `dofs` or `grid`. This class should be moved into `GridTools`.
